apps/cowswap-frontend/src/modules/twap/services/fetchTwapOrdersFromSafe.ts

```diff
@@ -62,11 +72,20 @@ async function fetchSafeTransactionsChunk(
     }
   }
 
-  return safeApiKit.getAllTransactions(safeAddress)
+  /**
+   * SafeApiKit set limit=20 by default and there is no way to change it using the SDK
+   *
+   */
+  const url = getSafeHistoryRequestUrl(chainId, safeAddress, 0)
```

Comment:
Shouldn't the offset be updated to match the current page you are on?

Reply:
No, we only do it one for the first request, for all the rest we take `next` URL from previous response.